### PR TITLE
added HW requirements

### PR DESCRIPTION
**Pull request by boroda4436**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,50 @@
...
+Add the implementation of `Single table` inheritance strategy.
+
+1. There are classes: Machine, Car and Truck. Both Car and Truck extends Machine. 
+Add the implementation of `Table per class` inheritance strategy.
+
+1. There are classes: Person, Coach and Mentor. Both Coach and Mentor extends Person. 
+Add the implementation of `Join tables` inheritance strategy.
+
+1. There are classes: Character, Bowman and Farmer. Both Bowman and Farmer extends Character. 
+Add the implementation of `Implicit strategy (Hibernate only)` inheritance strategy.
```
Comment (Sofasmile) [inline README.md:22]: ми хочемо щоб вони і це імплементовували? Як на мене то 3 основних - достатньо
ну і можна ще Mapped Superclass залишити

File: README.md [hunk 2]
```diff
@@ -1 +1,50 @@
...
+In this HW you will have a practice with different inheritance strategies.
+
+Action items:
+
+    - Clone the project. 
+    - Configure Hibernate. 
+    - Add required annotations in model classes. 
+    
+### Implement the following requirements: 
+1. There are classes: Animal, Cat and Dog. Both Cat and Dog extends Animal.
```
Comment (KseniiaMakarova) [inline README.md:12]:
```suggestion
1. There are classes: Animal, Cat and Dog. Both Cat and Dog extend Animal. 
```

File: README.md [hunk 3]
```diff
@@ -1 +1,50 @@
...
+
+    - Clone the project. 
+    - Configure Hibernate. 
+    - Add required annotations in model classes. 
+    
+### Implement the following requirements: 
+1. There are classes: Animal, Cat and Dog. Both Cat and Dog extends Animal. 
+Add the implementation of `Single table` inheritance strategy.
+
+1. There are classes: Machine, Car and Truck. Both Car and Truck extends Machine.
```
Comment (KseniiaMakarova) [inline README.md:15]:
```suggestion
1. There are classes: Machine, Car and Truck. Both Car and Truck extend Machine. 
```

File: README.md [hunk 4]
```diff
@@ -1 +1,50 @@
...
+    - Add required annotations in model classes. 
+    
+### Implement the following requirements: 
+1. There are classes: Animal, Cat and Dog. Both Cat and Dog extends Animal. 
+Add the implementation of `Single table` inheritance strategy.
+
+1. There are classes: Machine, Car and Truck. Both Car and Truck extends Machine. 
+Add the implementation of `Table per class` inheritance strategy.
+
+1. There are classes: Person, Coach and Mentor. Both Coach and Mentor extends Person.
```
Comment (KseniiaMakarova) [inline README.md:18]:
```suggestion
1. There are classes: Person, Coach and Mentor. Both Coach and Mentor extend Person. 
```

File: README.md [hunk 5]
```diff
@@ -1 +1,50 @@
...
+1. There are classes: Animal, Cat and Dog. Both Cat and Dog extends Animal. 
+Add the implementation of `Single table` inheritance strategy.
+
+1. There are classes: Machine, Car and Truck. Both Car and Truck extends Machine. 
+Add the implementation of `Table per class` inheritance strategy.
+
+1. There are classes: Person, Coach and Mentor. Both Coach and Mentor extends Person. 
+Add the implementation of `Join tables` inheritance strategy.
+
+1. There are classes: Character, Bowman and Farmer. Both Bowman and Farmer extends Character.
```
Comment (KseniiaMakarova) [inline README.md:21]:
```suggestion
1. There are classes: Character, Bowman and Farmer. Both Bowman and Farmer extend Character. 
```

File: README.md [hunk 6]
```diff
@@ -1 +1,50 @@
...
+1. There are classes: Machine, Car and Truck. Both Car and Truck extends Machine. 
+Add the implementation of `Table per class` inheritance strategy.
+
+1. There are classes: Person, Coach and Mentor. Both Coach and Mentor extends Person. 
+Add the implementation of `Join tables` inheritance strategy.
+
+1. There are classes: Character, Bowman and Farmer. Both Bowman and Farmer extends Character. 
+Add the implementation of `Implicit strategy (Hibernate only)` inheritance strategy.
+
+1. There are classes: Figure, Circle and Triangle. Both Circle and Triangle extends Figure.
```
Comment (KseniiaMakarova) [inline README.md:24]:
```suggestion
1. There are classes: Figure, Circle and Triangle. Both Circle and Triangle extend Figure. 
```

File: README.md [hunk 7]
```diff
@@ -1 +1,50 @@
...
+
+1. There are classes: Character, Bowman and Farmer. Both Bowman and Farmer extends Character. 
+Add the implementation of `Implicit strategy (Hibernate only)` inheritance strategy.
+
+1. There are classes: Figure, Circle and Triangle. Both Circle and Triangle extends Figure. 
+Add the implementation of `Mapped Superclass` inheritance strategy.
+
+### Implement the following methods
+1. Insert at least 5 dogs and 5 cats into the DB.
+1. Select all cats and dogs where name starts with 'A'.
```
Comment (KseniiaMakarova) [inline README.md:29]:
```suggestion
1. Select all cats and dogs whose name starts with 'A'.
```

File: README.md [hunk 8]
```diff
@@ -1 +1,50 @@
...
+1. Insert at least 3 bowmen and 5 farmers into the DB.
+1. Select all characters sorted ascending by power.
+1. Insert at least 3 circles and 5 triangles into the DB.
+1. Select all figures where color is red.
+
+### Embeddable
+1. There are classes: NewsPost and PostMetadata. Each NewsPost has its own PostMetadata. 
+Make PostMetadata embeddable in the NewsPost.
+1. Insert at least 5 posts with metadata.
+1. Select all cars and trucks older than 4 days.
```
Comment (KseniiaMakarova) [inline README.md:44]:
```suggestion
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: README.md [hunk 9]
```diff
@@ -1 +1,25 @@
...
+1. There are classes: Machine, Car and Truck. Both Car and Truck extend Machine. 
+Add the implementation of `Table per class` inheritance strategy.
+
+1. There are classes: Person, Coach and Mentor. Both Coach and Mentor extend Person. 
+Add the implementation of `Join tables` inheritance strategy.
+
+1. There are classes: Figure, Circle and Triangle. Both Circle and Triangle extend Figure. 
+Add the implementation of `Mapped Superclass` inheritance strategy.
+
+### Implement all required methods in DAOs
```
Comment (dmytro-hryhoruk) [inline README.md:25]: it seems to me that we had tasks here?

**Review by dmytro-hryhoruk — COMMENTED** [review]
I think we should add more information about dao methods and classess(one common, or for each entity e.g) base on our with @KseniiaMakarova  solutions

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**